ipaserver/install/dogtaginstance.py

```diff
@@ -294,24 +304,26 @@ def configure_certmonger_renewal(self):
                     # Give dogtag extra time to generate cert
                     timeout=CA_DBUS_TIMEOUT)
 
-    def __get_pin(self):
+    def __get_pin(self, token_name=None):
         try:
-            return certmonger.get_pin('internal')
+            return certmonger.get_pin(token_name)
```

Comment:
Rather than defaulting token_name to None why not default to "internal" then you can remove the test for None in get_pin() and other places? Theoretically internal:Server-Cert is the equivalent to Server-Cert so there shouldn't be any harm in always using internal unless you are trying to introduce as few changes as possible (it would certainly exercise some code in new ways).

Reply:
That was my initial implementation, too. But I ran into some issues with Dogtag. I may have identified the problem and may have found a workaround. Let's see if tests are now passing.